File: wrench-sample/src/main/java/com/example/wrench/SampleApplication.java
package com.example.wrench;

import android.content.Context;

import com.example.wrench.di.AppInjector;
import com.example.wrench.di.DaggerAppComponent;

import dagger.android.AndroidInjector;
import dagger.android.support.DaggerApplication;

public class SampleApplication extends DaggerApplication {
    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);

        AppInjector.init(this);
    }

    @Override
    protected AndroidInjector<? extends DaggerApplication> applicationInjector() {
        return DaggerAppComponent.builder().application(this).build();
    }
}

File: wrench-sample/src/main/java/com/example/wrench/SampleApplication.kt
package com.example.wrench

import android.app.Application
import com.example.wrench.di.sampleAppModule
import org.koin.android.ext.koin.androidContext
import org.koin.core.context.startKoin
import org.koin.dsl.koinApplication

class SampleApplication : Application() {

    override fun onCreate() {
        super.onCreate()

        startKoin(koinApplication {
            modules(listOf(sampleAppModule))
            androidContext(this@SampleApplication)
        })
    }
}